snapcraft/storeapi/__init__.py

```diff
@@ -437,6 +446,41 @@ def snap_release(self, snap_name, revision, channels):
 
         return response_json
 
+    def push_validation(self, snap_id, assertion):
+        data = {
+            'assertion': assertion.decode('utf-8'),
+        }
+        auth = _macaroon_auth(self.conf)
+        response = self.put(
+            'snaps/{}/validations'.format(snap_id), data=json.dumps(data),
+            headers={'Authorization': auth,
+                     'Content-Type': 'application/json',
+                     'Accept': 'application/json'})
+        if not response.ok:
+            raise errors.StoreValidationError(snap_id, response)
+        try:
+            response_json = response.json()
+        except (AttributeError, JSONDecodeError):
+            raise errors.StoreValidationError(snap_id, response)
```

Comment:
Honestly, not very good, like

```
Expecting value: line 1 column 1 (char 0)
```

I'd change it to a generic "Invalid response from server" maybe?